src/native/tools/malloc_napi.cpp

```diff
@@ -0,0 +1,54 @@
+/* Copyright (C) 2016 NooBaa */
+#include "../util/napi.h"
+#include "../util/common.h"
+
+#include <stdlib.h>
+#include <uv.h>
+#include <dlfcn.h>
+#include <malloc.h>
+
+namespace noobaa
+{
+
+DBG_INIT(0);
+
+static void (*malloc_stats_print)
+(
+    void (*write_cb) (void *, const char *), 
+    void *cbopaque, 
+    const char *opts
+) = 0;
+
+static void _print_malloc_stats(const Napi::CallbackInfo& info);
+static void _print_jemalloc_stats(const Napi::CallbackInfo& info);
+
+void
```

Comment:
I just wish there was a comment here or on the entire file, saying that his is for functions to the memory allocation system, and can print jemalloc stats (if loaded), or the default malloc stats otherwise.